integration/utils.go

```diff
@@ -14,11 +15,11 @@ func consumeAndLogOutput(c *check.C, id string, f io.ReadCloser, err error) {
 			f.Close()
 			c.Logf("Output %s: Closed", id)
 		}()
-		buf := make([]byte, 0, 1024)
+		buf := make([]byte, 1024)
 		for {
 			c.Logf("Output %s: waiting", id)
 			n, err := f.Read(buf)
-			c.Logf("Output %s: got %d,%#v: %#v", id, n, err, buf[:n])
+			c.Logf("Output %s: got %d,%#v: %s", id, n, err, strings.TrimSuffix(string(buf[:n]), "\n"))
```

Comment:
I’d rather preserve the output as much as possible (e.g. including leading whitespace for indentation, but mostly for the principle of the thing). The content of `c.Logf` is only shown to the user when the test fails, so this does not add any clutter in the common case.

The `TrimSuffix` is in there only so that _if_ the test fails and the log is printed, the log lines are not double-spaced (one newline per log item, one within the logged line in `buf`).